File: website/extract.py
import os
import json
import re
try:
    from PyPDF2 import PdfReader
except ImportError:
    print("PyPDF2 not found")
    exit(1)

dirs = [
    {'path': '../COSA SPORTS LISTS-MEN', 'gender': 'Men'},
    {'path': '../COSA SPORTS LISTS-WOMEN', 'gender': 'Women'}
]

result = []

for d in dirs:
    directory = d['path']
    if not os.path.exists(directory):
        print(f"Directory {directory} not found")
        continue
        
    for filename in os.listdir(directory):
        if not filename.endswith('.pdf'):
            continue
            
        filepath = os.path.join(directory, filename)
        
        try:
            reader = PdfReader(filepath)
            text = ""
            for page in reader.pages:
                text += page.extract_text() + "\n"
                
            lines = [line.strip() for line in text.split('\n') if line.strip()]
            
            # Clean up sport name
            sport_name = filename
            sport_name = re.sub(r'(?i)COSA', '', sport_name)
            sport_name = re.sub(r'(?i)MENS?', '', sport_name)
            sport_name = re.sub(r'(?i)WOMENS?', '', sport_name)
            sport_name = re.sub(r'(?i)\.pdf', '', sport_name)
            sport_name = sport_name.replace('.', '').strip()
            
            # Basic parsing of lines to extract names. The names are usually in lines after the header.
            # We'll just store the lines for now, or try to filter out Sr No, Name, Dept etc.
            names = []
            for line in lines:
                # If line is like "1 Soman Chalise TE"
                # Let's just capture anything that looks like a name.
                # Actually, storing the raw lines is safer so I can inspect.
                pass
                
            result.append({
                'sport': sport_name,
                'gender': d['gender'],
                'file': filename,
                'lines': lines
            })
        except Exception as e:
            print(f"Failed to read {filename}: {e}")

with open('extracted_sports.json', 'w', encoding='utf-8') as f:
    json.dump(result, f, indent=2)

print("Extraction complete")

File: website/extract.mjs
import fs from 'fs';
import path from 'path';
import pdf from 'pdf-parse';

const dirs = [
  { path: '../COSA SPORTS LISTS-MEN', gender: 'Men' },
  { path: '../COSA SPORTS LISTS-WOMEN', gender: 'Women' }
];

async function extract() {
  const result = [];
  
  for (const dir of dirs) {
    const files = fs.readdirSync(dir.path).filter(f => f.endsWith('.pdf'));
    for (const file of files) {
      const filePath = path.join(dir.path, file);
      const dataBuffer = fs.readFileSync(filePath);
      
      try {
        const data = await pdf(dataBuffer);
        // The text might have multiple lines, we can just split by line
        const lines = data.text.split('\n').map(l => l.trim()).filter(l => l);
        
        // Extracted sports name from filename: e.g. "COSA 1KM MEN..pdf" -> "1KM"
        let sportName = file.replace(/COSA/ig, '').replace(/MENS?/ig, '').replace(/WOMENS?/ig, '').replace(/\.pdf/ig, '').replace(/\./g, '').trim();
        
        result.push({
          sport: sportName,
          gender: dir.gender,
          file: file,
          lines: lines
        });
      } catch (err) {
        console.error('Failed to parse', file, err);
      }
    }
  }
  
  fs.writeFileSync('extracted_sports.json', JSON.stringify(result, null, 2));
  console.log('Extraction complete');
}

extract();
